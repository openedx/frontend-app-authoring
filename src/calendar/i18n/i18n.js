import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import Cookies from "universal-cookie";

import en from "./languages/en.json";
import hi from "./languages/hi.json";
import ar from "./languages/ar.json";
import fr from "./languages/fr.json";
import de from "./languages/de.json";
import ja from "./languages/ja.json";
import zh from "./languages/zh.json";

// import config from "../data/config.json";


const cookies = new Cookies();

const langFromCookie = cookies.get("openedx-language-preference") || "en";




const resources = {
  en: { translation: en },
  hi: { translation: hi },
  ar: { translation: ar },
  fr: { translation: fr },
  de: { translation: de },
  ja: { translation: ja },
  zh: { translation: zh }
};


i18n.use(initReactI18next).init({
  resources,
  lng: langFromCookie,  // âœ… string pass karna hai
  fallbackLng: "en",
  interpolation: { escapeValue: false }
});

// i18n.use(initReactI18next).init({
//   resources,
//   lng: config.language || "en",
//   fallbackLng: "en",
//   interpolation: { escapeValue: false }
// });

export default i18n;



