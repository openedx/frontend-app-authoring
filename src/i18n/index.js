import { messages as footerMessages } from '@edx/frontend-component-footer';
import { messages as paragonMessages } from '@edx/paragon';
import arMessages from './messages/ar.json';
import deMessages from './messages/de.json';
import deDEMessages from './messages/de_DE.json';
import es419Messages from './messages/es_419.json';
import faIRMessages from './messages/fa_IR.json';
import frMessages from './messages/fr.json';
import frCAMessages from './messages/fr_CA.json';
import hiMessages from './messages/hi.json';
import itMessages from './messages/it.json';
import itITMessages from './messages/it_IT.json';
import ptMessages from './messages/pt.json';
import ptPTMessages from './messages/pt_PT.json';
import ruMessages from './messages/ru.json';
import ukMessages from './messages/uk.json';
import zhCNMessages from './messages/zh_CN.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  de: deMessages,
  'de-de': deDEMessages,
  'es-419': es419Messages,
  'fa-ir': faIRMessages,
  fr: frMessages,
  'fr-ca': frCAMessages,
  hi: hiMessages,
  it: itMessages,
  'it-it': itITMessages,
  ru: ruMessages,
  pt: ptMessages,
  'pt-pt': ptPTMessages,
  uk: ukMessages,
  'zh-cn': zhCNMessages,
};

export default [
  footerMessages,
  paragonMessages,
  appMessages,
];
