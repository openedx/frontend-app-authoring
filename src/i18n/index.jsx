import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  'es-419': es419Messages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
};

export default messages;
