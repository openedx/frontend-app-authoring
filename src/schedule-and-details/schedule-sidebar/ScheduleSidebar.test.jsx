import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';
import { AppContext } from '@edx/frontend-platform/react';

import ScheduleSidebar from '.';

const mockPathname = '/foo-bar';
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: mockPathname,
  }),
}));

describe('<ScheduleSidebar />', () => {
  const config = { STUDIO_BASE_URL: 'https://example.com' };
  const courseId = 'course123';
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        // eslint-disable-next-line react/jsx-no-constructed-context-values
        <AppContext.Provider value={{ config }}>
          <IntlProvider locale="en">
            <ScheduleSidebar
              intl={injectIntl}
              courseId={courseId}
              proctoredExamSettingsUrl="link://to"
            />
          </IntlProvider>
        </AppContext.Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
