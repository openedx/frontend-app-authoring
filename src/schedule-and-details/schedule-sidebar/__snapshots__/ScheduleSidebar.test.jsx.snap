// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScheduleSidebar /> should match the snapshot 1`] = `
<aside
  className="help-sidebar"
>
  <div
    className="help-sidebar-about"
  >
    <h4
      className="help-sidebar-about-title"
    >
      How are these settings used?
    </h4>
    <p
      className="help-sidebar-about-descriptions"
    >
      Your course's schedule determines when students can enroll in and begin a course. Other information from this page appears on the About page for your course. This information includes the course overview, course image, introduction video, and estimated time requirements. Students use About pages to choose new courses to take.
    </p>
  </div>
  <hr />
  <div
    className="help-sidebar-other"
  >
    <h4
      className="help-sidebar-other-title"
    >
      Other Course Settings
    </h4>
    <nav
      aria-label="Other Course Settings"
      className="help-sidebar-other-links"
    >
      <ul
        className="p-0 mb-0"
      >
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="/course/course123/settings/details"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Schedule & Details
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="/course/course123/settings/grading"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Grading
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="/course/course123/course_team"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Course Team
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="link://to"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Proctored exam settings
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="https://example.com/group_configurations/course123"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Group Configurations
          </a>
        </li>
        <li
          className="help-sidebar-other-link"
        >
          <a
            className="pgn__hyperlink default-link standalone-link"
            href="/course/course123/settings/advanced"
            onClick={[Function]}
            rel="noopener"
            target="_self"
          >
            Advanced Settings
          </a>
        </li>
      </ul>
    </nav>
  </div>
</aside>
`;
